tests/builder/help.rs

```diff
@@ -2820,3 +2820,71 @@ fn display_name_subcommand_explicit() {
         Some("child.display")
     );
 }
+
+#[test]
+fn value_terminator_in_usage_message() {
+    static EXPECTED: &str = "\
+Usage: cmd [<multiple_options>... :] [last]
+
+Arguments:
+  [<multiple_options>... :]  
+  [last]                     
+
+Options:
+  -h, --help  Print help
+";
+    let cmd = Command::new("cmd")
+        .arg(
+            Arg::new("multiple_options")
+                .num_args(0..)
+                .value_terminator(":"),
+        )
+        .arg(Arg::new("last").required(false));
+    utils::assert_output(cmd, "cmd --help", EXPECTED, false);
+}
+
+#[test]
+fn non_positional_value_terminator_in_usage_message() {
+    static EXPECTED: &str = "\
+Usage: cmd [OPTIONS] [last]
+
+Arguments:
+  [last]  
+
+Options:
+  -o, --multiple_options [<multiple_options>... :]  
+  -h, --help                                        Print help
+";
+    let cmd = Command::new("cmd")
+        .arg(
+            Arg::new("multiple_options")
+                .short('o')
+                .long("multiple_options")
+                .num_args(0..)
+                .value_terminator(":"),
+        )
+        .arg(Arg::new("last").required(false));
+    utils::assert_output(cmd, "cmd --help", EXPECTED, false);
+}
+
+#[test]
+fn required_options_with_value_terminator_in_usage_message() {
+    static EXPECTED: &str = "\
+Usage: cmd [<multiple_options>... :] [<multiple_options>... :] [<multiple_options>... :] [last]
+
+Arguments:
+  [<multiple_options>... :] [<multiple_options>... :] [<multiple_options>... :]  
+  [last]                                                                         
+
+Options:
+  -h, --help  Print help
+";
+    let cmd = Command::new("cmd")
+        .arg(
+            Arg::new("multiple_options")
+                .num_args(3..)
+                .value_terminator(":"),
+        )
```

Comment:
What about a test where `multiple_options` is required?

We also should have two error tests
- Missing required argument (to verify how it shows up in the "required args" list)
- Argument is present but another argument has an error (to verify how it shows up in the modified usage for errors)